sympy/functions/special/bessel.py

```diff
@@ -216,11 +217,40 @@ def _eval_rewrite_as_bessely(self, nu, z, **kwargs):
     def _eval_rewrite_as_jn(self, nu, z, **kwargs):
         return sqrt(2*z/pi)*jn(nu - S.Half, self.argument)
 
+    def _eval_as_leading_term(self, x, cdir=0):
+        nu, z = self.args
+        arg = z.as_leading_term(x)
+        if x in arg.free_symbols:
+            return z**nu
+        else:
+            return self.func(*self.args)
+
     def _eval_is_extended_real(self):
         nu, z = self.args
         if nu.is_integer and z.is_extended_real:
             return True
 
+    def _eval_nseries(self, x, n, logx, cdir=0):
+        from sympy.series.order import Order
+        nu, z = self.args
+
+        # In case of powers less than 1, number of terms need to be computed
+        # separately to avoid repeated callings of _eval_nseries with wrong n
+        try:
+            _, exp = z.leadterm(x)
+        except (ValueError, NotImplementedError):
+            return self
+
+        if exp.is_positive:
+            newn = ceiling(n/exp)
+            s = [(-1)**k * z**(2 * k) / (factorial(k) * factorial(nu + k)
+                    * 2**(2 * k)) for k in range((newn + 1) // 2)]
+            r = [k._eval_nseries(x, n, logx, cdir) for k in s]
+            t = z._eval_nseries(x, n, logx, cdir)
```

Comment:
`z._eval_nseries` will be called repeatedly on the line above (for each `z**(2*k)`). Would it be more efficient to compute it once and use the result (without the order term) in `s`?

Reply:
I see my mistake, the method you suggested would definitely be much faster.

Reply:
Maybe we should first compute the expansion of `z**2`, and then iteratively `z**(2*k + 2) = z**(2*k)*z**2`.

Reply:
I didn't fully understand what you meant. Are you referring to something like this?
```py
    newn = ceiling(n/exp)                                                       
    r = z._eval_nseries(x, n, logx, cdir).removeO()                             
    t = (z**2)._eval_nseries(x, n, logx, cdir).removeO()                        
    s = []                                                                      
    p = 1                                                                       
    for k in range((newn + 1)//2):                                              
        term = (-1)**k * p / (factorial(k) * factorial(nu + k)* 2**(2 * k))     
        s.append(term)                                                          
        p *= t                                                                  
    return ((r / 2)**nu * Add(*s)).expand() + Order(x**n, x)                    
```

Reply:
Yes, something like that.